packages/big-design/src/components/Counter/Counter.tsx

```diff
@@ -0,0 +1,214 @@
+      }
+      // Checks that the provided value is a multiple of the step
+      if (value % step === 0) {
+        onCountChange(value - step);
+      }
+      // If not, returns nearest lower value that is a multiple of the step
+      else {
+        onCountChange(value - (value % step));
+      }
+    };
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      const newValue = Number(event.currentTarget.value);
+      if (isNaN(newValue)) {
+        return;
+      }
+      if (newValue >= min && newValue <= max) {
+        onCountChange(newValue);
+      }
+
+      return;
+    };
+
+    const handleKeyPress = (event: React.KeyboardEvent) => {
+      switch (event.key) {
+        case 'ArrowUp':
+          handleIncrease();
+          break;
+        case 'ArrowDown':
+          handleDecrease();
+          break;
+        default:
+          break;
+      }
+    };
+
+    const renderedLabel = useMemo(() => {
+      if (!label) {
+        return null;
+      }
+
+      if (typeof label === 'string') {
+        return (
+          <FormControlLabel id={labelId} htmlFor={id} renderOptional={!props.required}>
+            {label}
+          </FormControlLabel>
+        );
+      }
+
+      if (React.isValidElement(label) && label.type === FormControlLabel) {
+        return React.cloneElement(label as React.ReactElement<React.LabelHTMLAttributes<HTMLLabelElement>>, {
+          id: labelId,
+          htmlFor: id,
+        });
+      }
+
+      warning('label must be either a string or a FormControlLabel component.');
+    }, [id, label, labelId, props.required]);
+
+    const renderedDescription = useMemo(() => {
+      if (!description) {
+        return null;
+      }
+
+      if (typeof description === 'string') {
+        return <FormControlDescription>{description}</FormControlDescription>;
+      }
+
+      if (React.isValidElement(description) && description.type === FormControlDescription) {
+        return description;
+      }
+
+      warning('description must be either a string or a FormControlDescription component.');
+    }, [description]);
+
+    const errors = useMemo(() => {
+      const validateError = (err: CounterProps['error']) => {
+        if (!err) {
+          return null;
+        }
+
+        if (typeof err === 'string') {
+          return err;
+        }
+
+        if (React.isValidElement(err) && err.type === FormControlError) {
+          return err;
+        }
+
+        warning('error must be either a string or a FormControlError component.');
+      };
+
+      if (Array.isArray(error)) {
+        error.forEach(validateError);
+
+        return error;
+      }
+
+      return validateError(error);
+    }, [error]);
+
+    return (
+      <div>
+        {renderedLabel}
+        {renderedDescription}
+        <StyledCounterWrapper disabled={disabled} error={errors} focus={focus}>
+          <RemoveCircleOutlineIcon onClick={handleDecrease} color={decreaseIconColor()} />
```

Comment:
One thing I noticed about these icons is that you can't tab over them/handle keyboard events. To kind of simplify things, you could wrap these in an extended `StyleableButton`, and you'll get baked in accessibility.

Reply:
We'll probably also want a `title` on the SVGs.